app/scripts/controllers/transactions/tx-state-manager.js

```diff
@@ -279,9 +272,8 @@ export default class TransactionStateManager extends EventEmitter {
     const txsToDelete = transactions
       .reverse()
       .filter((tx) => {
-        const { nonce, from } = tx.txParams;
-        const { chainId, metamaskNetworkId, status } = tx;
-        const key = `${nonce}-${chainId ?? metamaskNetworkId}-${from}`;
+        const { chainId, status, txParams } = tx;
+        const key = `${txParams.nonce}-${chainId}-${txParams.from}`;
```

Comment:
It seems like this would be one of the most impactful lines in this PR. What are the possible effects of this change?

Reply:
if there are greater than 60 of transactions that do not have `txMeta.chainId` populated, only the most recent 60 of those transactions will be kept. The older ones are purged from state. This would mirror the new behavior in [the core TransactionController ](https://github.com/MetaMask/core/blob/5e1044cd02180ad9bfe574a9ec4096b13ca0efbd/packages/transaction-controller/src/TransactionController.ts#L1379C17-L1381).

Reply:
Makes sense!